packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -0,0 +1,217 @@
+   */
+  columnHeaderHeight: PropTypes.string,
+  /**
+   * String that specifies the default height for rows in the grid. Any valid CSS width value is accepted.
+   * To override the default value, provide height for the row that needs to be overridden.
+   */
+  rowHeight: PropTypes.string,
+  /**
+   * Number indicating the index of the column that represents row header. Index is 0 based and cannot exceed one less than the number of columns in the grid.
+   */
+  rowHeaderIndex: PropTypes.number,
+};
+
+const defaultProps = {
+  rowHeaderIndex: 0,
+};
+
+function WorklistDataGrid(props) {
+  const {
+    id,
+    ariaLabelledby,
+    ariaLabel,
+    columns,
+    rows,
+  } = props;
+
+  const focusedRow = useRef(0);
+  const focusedCol = useRef(0);
+  const grid = useRef();
+
+  const gridRef = useCallback((node) => {
+    grid.current = node;
+    const focusedCell = grid.current.rows[focusedRow.current].cells[focusedCol.current];
+    focusedCell.tabIndex = 0;
+  }, []);
+
+  const handleKeyDown = (event) => {
+    const currentFocusedRow = event.target.parentElement.rowIndex;
+    const currentFocusedCol = event.target.cellIndex;
+
+    let nextRow = currentFocusedRow;
+    let nextCol = currentFocusedCol;
+
+    const key = event.keyCode;
+    switch (key) {
+      case KeyCode.KEY_UP:
+        nextRow -= 1;
+        break;
+      case KeyCode.KEY_DOWN:
+        nextRow += 1;
+        break;
+      case KeyCode.KEY_LEFT:
+        nextCol -= 1;
+        break;
+      case KeyCode.KEY_RIGHT:
+        nextCol += 1;
+        break;
+      case KeyCode.KEY_HOME:
+        nextCol = 0;
+        if (event.ctrlKey) {
+          nextRow = 1; // Assumption is that the first row is the column Heading.
+        }
+        break;
+      case KeyCode.KEY_END:
+        nextCol = props.rows[0].cells.length - 1;
+        if (event.ctrlKey) {
+          nextRow = props.rows.length;
+        }
+        break;
+      default:
+        return;
+    }
+    if (nextRow >= grid.current.rows.length || nextCol >= grid.current.rows[0].cells.length) {
+      event.preventDefault();
+      return;
+    }
+    if (nextCol < 0 || nextRow < 0) {
+      event.preventDefault();
+      return;
+    }
+
+    // Remove the ability to receive visible focus as only one cell should be able to receive tab focus.
+    // This also ensures that tabbing back into the grid focuses on the cell that last had focus.
+    grid.current.rows[currentFocusedRow].cells[currentFocusedCol].tabIndex = -1;
+
+    focusedRow.current = nextRow;
+    focusedCol.current = nextCol;
+
+    const elementWithTabStop = grid.current.rows[nextRow].cells[nextCol];
+    elementWithTabStop.tabIndex = 0;
+    elementWithTabStop.focus();
+    event.preventDefault();
+  };
+
+  const handleClick = (event) => {
+    const clickedCell = event.target.closest('td,th') || document.activeElement.closest('td,th');
+    if (!clickedCell) {
+      event.preventDefault();
```

Comment:
Why are we preventing the default behavior in this scenario?

Reply:
I think it can be removed. We will test without it.